### PR TITLE
Python 3.13 Support

### PR DESCRIPTION
**Pull request by ulgens**
### What are you trying to accomplish?

Add Python 3.13 support and bump pyenv version.

Resolves https://github.com/dependabot/dependabot-core/issues/10828

I tried to copy related changes from https://github.com/dependabot/dependabot-core/pull/8732 but I feel like I missed if there is anything special required for 3.13 . Please feel free to direct me whatever needs to be added or fixed.

### Checklist
- [ ] I have run the complete test suite to ensure all tests and linters pass.
- [ ] I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
- [x] I have written clear and descriptive commit messages.
- [x] I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [x] I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

**Comment by ulgens** [comment]
@jurre @landongrindheim @deivid-rodriguez Hey all 👋🏻 I want to finish this piece of work and push it forward but I don't understand why `ci (python, python_slow, pip)` pipeline fails.

**Comment by edgarrmondragon** [comment]
@ulgens I think it's trying to install a Pandas version that fails to build on Python 3.13. Might be worth bumping this

https://github.com/dependabot/dependabot-core/blob/313fcff149b3126cb78b38d15f018907d729f8cc/python/spec/fixtures/requirements/pip_compile_native_dependencies.txt#L12

**Comment by ulgens** [comment]
It seems the issue is https://github.com/librosa/librosa/issues/1883 and it's waiting for a new release of numba. Numba release is blocked by https://github.com/numba/numba/milestone/81 .

@feelepxyz Is there a specific reason to have `librosa` in `python/spec/fixtures/projects/unresolvable/requirements.in`? Can we remove or replace it?

**Comment by zhu0629** [comment]
why dependabot use `librosa`? for what?

**Comment by edgarrmondragon** [comment]
> Any update on this?

AFAICT, the dependabot team is swamped so it's probably take them some time to review this as I don't think it's really a priority or a blocker for a lot of people.

**Comment by jarey** [comment]
Numba 0.61.0rc1 could potentially umblock this.

**Comment by jeffwidman** [comment]
Oof, thanks @ulgens let's seee if we can get this out the door.

**Comment by jeffwidman** [comment]
> Is there a specific reason to have librosa in python/spec/fixtures/projects/unresolvable/requirements.in? Can we remove or replace it?

@ulgens it's replacable/removable... the point of that test is described in its original commit: https://github.com/dependabot/dependabot-core/commit/4ff9a8efcc432b9c0bea80b043430139d88610e1

So as long as we continue to replicate that situation, we can swap out the individual deps as needed. As visible in [the commit history for that file](https://github.com/dependabot/dependabot-core/commits/main/python/spec/fixtures/projects/unresolvable/requirements.in), it's been a source of friction in past major python upgrade as well. So it'd be nice if we could swap to a set of deps that aren't so brittle... I suspect that numpy/pandas/numba all have enough `C` in them that they are more work to upgrade, vs a set of pure python deps would _just work_ with the new major version. This isn't a blocker for this PR, more just an observation.

File: python/helpers/requirements.txt [hunk 1]
```diff
@@ -2,6 +2,8 @@ pip==24.0
 pip-tools==7.4.1
 flake8==7.1.0
 hashin==1.0.1
+# TODO: Remove "legacy-cgi" dependency when hashin is updated to support Python 3.13
```
Comment (jeffwidman) [inline python/helpers/requirements.txt:5]:
```suggestion
# TODO: Remove "legacy-cgi" dependency when we update `hashin` to v1.0.3 which supports Python 3.13.
# But `hashin` v1.0.3 also dropped support for Python 3.8 so first we have to drop 3.8 support in Dependabot.
```

**Comment by randhircs** [comment]
@ulgens Merging is blocked since one Required Specs is  failing, you may like to fix that.

File: python/helpers/requirements.txt (hunk 1 above)
Comment (jeffwidman) [reply python/helpers/requirements.txt:5]: True, I'd forgotten about conditional imports.

Comment (ulgens) [reply python/helpers/requirements.txt]: Thanks to https://github.com/dependabot/dependabot-core/pull/11115 getting merged, this conditional dependency is no longer required. I destroyed the commit.

**Comment by sainak** [comment]
@ulgens the spec is failing because it's trying to install pandas v2.1.1 on py3.13, it should work fine with pandas>=2.2.0

**Comment by ulgens** [comment]
Not sure how I pulled librosa into this and missed pandas, but the problem was pandas and it's okay now. Thanks @edgarrmondragon @daveisfera and @sainak

**Review by amazimbe — APPROVED** [review]

**Review by amazimbe — APPROVED** [review]

**Comment by ulgens** [comment]
@amazimbe Thank you so much for the approval and the merge 🌻

**Pull request merged**